clang/lib/CIR/CodeGen/CIRGenBuiltinAArch64.cpp

```diff
@@ -30,6 +31,27 @@ using namespace clang;
 using namespace clang::CIRGen;
 using namespace llvm;
 
+template <typename... Operands>
+static mlir::Value emitIntrinsicCallOp(CIRGenBuilderTy &builder,
+                                       mlir::Location loc, const StringRef str,
+                                       const mlir::Type &resTy,
+                                       Operands &&...op) {
+  return cir::LLVMIntrinsicCallOp::create(builder, loc,
+                                          builder.getStringAttr(str), resTy,
+                                          std::forward<Operands>(op)...)
+      .getResult();
+}
+
+// Generate vscale * scalingFactor
+static mlir::Value genVscaleTimesFactor(mlir::Location loc,
+                                        CIRGenBuilderTy builder,
+                                        mlir::Type cirTy,
+                                        int32_t scalingFactor) {
+  auto vscale = emitIntrinsicCallOp(builder, loc, "vscale", cirTy);
```

Comment:
```suggestion
  mlir::Value vscale = emitIntrinsicCallOp(builder, loc, "vscale", cirTy);
```
We're much less liberal with `auto` in Clang than MLIR code is.